wrpc/worker.go

```diff
@@ -0,0 +1,149 @@
+package wrpc
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	pb "github.com/bgzzz/go-schedule/proto"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type Worker struct {
+	id           string
+	stopSender   chan struct{}
+	send         chan interface{}
+	streamServer pb.Scheduler_WorkerConnectServer
+	streamClient pb.Scheduler_WorkerConnectClient
+
+	err      error
+	errorMtx sync.RWMutex
+
+	silenceTimeout time.Duration
+}
+
+func (w *Worker) StartSender() {
+	go func() {
+		for {
+			select {
+			case <-w.stopSender:
+				{
+					log.Debugf("sender is stopped for worker node ")
+					return
+				}
+			case msg := <-w.send:
+				{
+					log.Debugf("tx: %+v", msg)
+					err := w.SendWithTimeout(msg)
+					if err != nil {
+						log.Error(err.Error())
+						w.SetErr(err)
+					}
+				}
+			}
+		}
+	}()
+}
+
+func (w *Worker) StopSender() {
+	w.stopSender <- struct{}{}
+	close(w.stopSender)
+}
+
+func (w *Worker) CheckErr() error {
+	w.errorMtx.RLock()
+	defer w.errorMtx.RUnlock()
+
+	err := w.err
+	return err
+}
+
+func (w *Worker) SetErr(err error) {
+	w.errorMtx.Lock()
+	defer w.errorMtx.Unlock()
+
+	if w.err != nil {
+		w.err = fmt.Errorf("%s %s", w.err, err)
+	}
+}
+
+// RxMsg is helper struct
+type rxMsg struct {
+	err error
+	msg interface{}
+}
+
+func (w *Worker) RxWithTimeout() (interface{}, error) {
+	msgChan := make(chan rxMsg, 1)
+	go func() {
+		var msg interface{}
+		var err error
+		if w.streamServer != nil {
+			msg, err = w.streamServer.Recv()
+		} else {
+			msg, err = w.streamClient.Recv()
+		}
+		msgChan <- rxMsg{
+			err: err,
+			msg: msg,
+		}
+		close(msgChan)
+	}()
+	t := time.NewTimer(w.silenceTimeout * time.Second)
+	select {
+	case <-t.C:
```

Comment:
Timer will be leaked here. I'd generally do `defer t.Stop()` right after timer is created to avoid this.